parse_cutadapt_logs.py

```diff
@@ -0,0 +1,174 @@
+__credits__ = ["Gavin Douglas"]
+__license__ = "GPL"
+__version__ = "1.0"
+__maintainer__ = "Gavin Douglas"
+__email__ = "gavin.douglas@dal.ca"
+__status__ = "Development"
+
+import argparse
+import re
+import os
+
+parser = argparse.ArgumentParser(description="Parse cutadapt logfiles to \
+produce a table of read counts per sample.", epilog='''Usage example:
+
+parse_cutadapt_logs.py -i primer_trimmed_fastqs -m "_log.txt" \
+-o cutadapt_log.txt
+
+''', formatter_class=argparse.RawDescriptionHelpFormatter)
+
+parser.add_argument("-i", "--input", help="Folder containing cutadapt \
+logfiles", required=True, type=str)
+
+parser.add_argument("-m", "--match", help="String to match cutadapt log \
+filenames", required=False, default="_log.txt", type=str)
+
+parser.add_argument("-s", "--sample_delim", help="String to split filename on \
+to determine sample name (assumed to be 1st field after splitting",
+                    required=False, default="_", type=str)
+
+parser.add_argument("-o", "--output", help="Name of output read count table",
+                    required=False, default="cutadapt_log.txt", type=str)
+
+
+def main():
+
+    args = parser.parse_args()
+
+    # Check whether input folder exists.
+    if not os.path.isdir(args.input):
+        raise Exception("Input path " + args.input + " is not a directory")
+
+    # Read in cutadapt log filenames.
+    logfiles = [f for f in os.listdir(args.input) if args.match in f]
+
+    # Throw exception if no logfiles matched string.
+    if len(logfiles) == 0:
+        raise Exception("No logfiles found in folder " +
+                        args.input + " matching string " + args.match)
+
+    # Open output file for writing.
+    outfile = open(args.output, "w")
+
+    # Put samples in set to double-check they are all unique.
+    past_samples = set()
+
+    # Set flag for whether header printed or not.
+    header_printed = False
+
+    # Loop over all logfiles.
+    for logfile in logfiles:
+
+        sample = logfile.split(args.sample_delim)[0]
+
+        # Check whether sample already seen and if not add to set.
+        if sample in past_samples:
+            raise Exception("Parsed sample name " + sample +
+                            " is not unique!")
+
+        past_samples.add(sample)
+
+        log = args.input + "/" + logfile
+
+        # Initialize library object which will be either "single" or "paired"
+        library = None
+
+        # Initialize empty dict which will be actual counts parsed.
+        read_counts = {}
+
+        with open(log, "r") as infile:
+
+            for line in infile:
+
+                # Check if library defined. If not then check if library info
+                # is on line. If not then skip line. Set strings that should
+                # be matched as well depending on library type and what
+                # categories these strings correspond to.
+                if not library:
+                    if "in paired-end mode ..." in line:
```

Comment:
Is this line the first one, or at least the last one before the summary counts? If so, you could move this whole check above `for line in infile` and handle the library checking in one section before moving on to counting things.

Reply:
That would probably be slightly more efficient to check for the library anywhere in the file before reading line-by-line, but I'm going to leave it since it's such a fast running script anyway.